File: Examples/VisualBasic/WorkingWithResourceAssignments/GenerateResourceAssignmentTimephasedData.vb

Namespace WorkingWithResourceAssignments
    Public Class GenerateResourceAssignmentTimephasedData
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' ExStart:GenerateResourceAssignmentTimephasedData
            ' Create project instance
            Dim project1 As New Project(dataDir + "ResourceAssignmentTimephasedData.mpp")

            ' Get the first task of the Project
            Dim task As Task = project1.RootTask.Children.GetById(1)

            ' Get the First Resource Assignment of the Project
            Dim firstRA As ResourceAssignment = project1.ResourceAssignments.ToList()(0)

            ' Flat contour is default contour
            Console.WriteLine("Flat contour")
            Dim td As TimephasedData
            Dim tdList = task.GetTimephasedData(project1.Get(Prj.StartDate), project1.Get(Prj.FinishDate))
            For Each td In tdList
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value)
            Next

            ' Change contour
            firstRA.Set(Asn.WorkContour, WorkContourType.Turtle)
            Console.WriteLine("Turtle contour")
            tdList = task.GetTimephasedData(project1.Get(Prj.StartDate), project1.Get(Prj.FinishDate))
            For Each td In tdList
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value)
            Next

            ' Change contour
            firstRA.Set(Asn.WorkContour, WorkContourType.BackLoaded)
            Console.WriteLine("BackLoaded contour")
            tdList = task.GetTimephasedData(project1.Get(Prj.StartDate), project1.Get(Prj.FinishDate))
            For Each td In tdList
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value)
            Next

            ' Change contour
            firstRA.Set(Asn.WorkContour, WorkContourType.FrontLoaded)
            Console.WriteLine("FrontLoaded contour")
            tdList = task.GetTimephasedData(project1.Get(Prj.StartDate), project1.Get(Prj.FinishDate))
            For Each td In tdList
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value)
            Next

            ' Change contour
            firstRA.Set(Asn.WorkContour, WorkContourType.Bell)
            Console.WriteLine("Bell contour")
            tdList = task.GetTimephasedData(project1.Get(Prj.StartDate), project1.Get(Prj.FinishDate))
            For Each td In tdList
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value)
            Next

            ' Change contour
            firstRA.Set(Asn.WorkContour, WorkContourType.EarlyPeak)
            Console.WriteLine("EarlyPeak contour")
            tdList = task.GetTimephasedData(project1.Get(Prj.StartDate), project1.Get(Prj.FinishDate))
            For Each td In tdList
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value)
            Next

            ' Change contour
            firstRA.Set(Asn.WorkContour, WorkContourType.LatePeak)
            Console.WriteLine("LatePeak contour")
            tdList = task.GetTimephasedData(project1.Get(Prj.StartDate), project1.Get(Prj.FinishDate))
            For Each td In tdList
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value)
            Next

            ' Change contour
            firstRA.Set(Asn.WorkContour, WorkContourType.DoublePeak)
            Console.WriteLine("DoublePeak contour")
            tdList = task.GetTimephasedData(project1.Get(Prj.StartDate), project1.Get(Prj.FinishDate))
            For Each td In tdList
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value)
            Next
            ' ExEnd:GenerateResourceAssignmentTimephasedData
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithResourceAssignments/GenerateResourceAssignmentTimephasedData.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithResourceAssignments
{
    using System;

    internal class GenerateResourceAssignmentTimephasedData
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());

            //ExStart:GenerateResourceAssignmentTimephasedData
            //ExFor: Asn.WorkContour
            //ExSummary: Shows how to set different timephased data contours for resource assignments.
            var project = new Project(dataDir + "ResourceAssignmentTimephasedData.mpp");

            // Get the first task of the Project
            var task = project.RootTask.Children.GetById(1);

            // Get the First Resource Assignment of the Project
            var firstRA = project.ResourceAssignments.ToList()[0];

            // Flat contour is default contour
            Console.WriteLine("Flat contour");

            var collection = task.GetTimephasedData(project.Get(Prj.StartDate), project.Get(Prj.FinishDate));
            foreach (var td in collection)
            {
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value);
            }

            // Change contour
            Console.WriteLine("Turtle contour");
            firstRA.Set(Asn.WorkContour, WorkContourType.Turtle);
            collection = task.GetTimephasedData(project.Get(Prj.StartDate), project.Get(Prj.FinishDate));
            foreach (var td in collection)
            {
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value);
            }

            // Change contour
            Console.WriteLine("BackLoaded contour");
            firstRA.Set(Asn.WorkContour, WorkContourType.BackLoaded);
            collection = task.GetTimephasedData(project.Get(Prj.StartDate), project.Get(Prj.FinishDate));
            foreach (var td in collection)
            {
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value);
            }

            // Change contour
            Console.WriteLine("FrontLoaded contour");
            firstRA.Set(Asn.WorkContour, WorkContourType.FrontLoaded);
            collection = task.GetTimephasedData(project.Get(Prj.StartDate), project.Get(Prj.FinishDate));
            foreach (var td in collection)
            {
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value);
            }

            // Change contour
            Console.WriteLine("Bell contour");
            firstRA.Set(Asn.WorkContour, WorkContourType.Bell);
            collection = task.GetTimephasedData(project.Get(Prj.StartDate), project.Get(Prj.FinishDate));
            foreach (var td in collection)
            {
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value);
            }

            // Change contour
            Console.WriteLine("EarlyPeak contour");
            firstRA.Set(Asn.WorkContour, WorkContourType.EarlyPeak);
            collection = task.GetTimephasedData(project.Get(Prj.StartDate), project.Get(Prj.FinishDate));
            foreach (var td in collection)
            {
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value);
            }

            // Change contour
            Console.WriteLine("LatePeak contour");
            firstRA.Set(Asn.WorkContour, WorkContourType.LatePeak);
            collection = task.GetTimephasedData(project.Get(Prj.StartDate), project.Get(Prj.FinishDate));
            foreach (var td in collection)
            {
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value);
            }
            
            // Change contour
            Console.WriteLine("DoublePeak contour");
            firstRA.Set(Asn.WorkContour, WorkContourType.DoublePeak);
            collection = task.GetTimephasedData(project.Get(Prj.StartDate), project.Get(Prj.FinishDate));
            foreach (var td in collection)
            {
                Console.WriteLine(td.Start.ToShortDateString() + " " + td.Value);
            }
            //ExEnd:GenerateResourceAssignmentTimephasedData
        }
    }
}